home/tests/test_add_answer.py

```diff
@@ -0,0 +1,50 @@
+from home.models import Answer
+import pytest
+
+
+@pytest.mark.django_db
+class TestAddAnswer:
+    @pytest.fixture
+    def comment_content(self):
+        return {'content': 'test'}
+
+    def test_post_redirect(self, client, question_test_data, authenticated_user, comment_content):
+        response = client.post(f'/explore/question_{question_test_data.id}/', data=comment_content)
+        assert (response.status_code == 302)  # check that redirect
+        # redirect back to the same display question page
+        assert (response.url == f'/explore/question_{question_test_data.id}/')
+
+    def test_add_answer_post(self, authenticated_user, client, question_test_data, comment_content):
+        client.post(f'/explore/question_{question_test_data.id}/', data=comment_content)
+        answers_feed = question_test_data.get_answers_feed()
+        assert (answers_feed.first().id)  # check if there is answer to the question_test_data
+        assert (answers_feed.first().content == 'test')  # answer content saved as expected.
+        assert (answers_feed.first().profile.user == authenticated_user)  # answer profile is the user that posted.
+
+    def test_html_authenticated_user_form_visible(self, authenticated_user, question_test_data, client):
+        response = client.get(f'/explore/question_{question_test_data.id}/')
+        # comment-form is the div id of the comment-form, should be visible (because it is authenticated user)
+        assert "comment-form" in str(response.content)
+        # not-authenticated is the div for login button for not authenticated user, should be hidden
+        assert "not-authenticated" not in str(response.content)
+
+    def test_html_not_authenticated_user_form_visible(self, question_test_data, client):
+        response = client.get(f'/explore/question_{question_test_data.id}/')
+        # should be hidden for not authenticated user
+        assert "comment-form" not in str(response.content)
+        # login button for not authenticated user should be visible
+        assert "not-authenticated" in str(response.content)
```

Comment:
Missing test for unauthenticated user trying to post

Reply:
added